docs/devel/networking.md

```diff
@@ -56,7 +56,10 @@ the following operations, in this order:
 The virt-launcher is an untrusted component of KubeVirt (since it wraps the
 libvirt process that will run third party workloads). As a result, it must be
 run with as little privileges as required. As of now, the only capability
-required by virt-launcher to configure networking is the `CAP_NET_ADMIN`.
+required by virt-launcher to configure networking is `CAP_NET_BIND_SERVICE` -
+and **only** when the VMs are **either** running as a non-root user **or**
```

Comment:
Can you please explain me how the non-root user is related?

Reply:
I think you require the capability also with root but it's not explicit.

Reply:
> For example VM with brige binding with no ipam (doesn't require a DHCP server).
It requires CAP_NET_BIND_SERVICE when the VM is running as a non-root user, but doesn't require it when the VM is running as root?

IMO, this is a different thing. You wouldn't *ever* need to bind to a low port when you're not using DHCP on the interface. But, I don't think you can actually (without a major refactor) understand when you're templating the pod if you'll have DHCP in the iface or not. I.e. you would need to read the net-attach-def and check if it has IPAM setting or not. 

Regarding the non-root user, I'm just reading the code. AFAIU, you make a good question - so good it should have been made when the production code was written.

@AlonaKaplan

Reply:
> Regarding the non-root user, I'm just reading the code
@maiqueb can you please send a link to the code you're reading?

Reply:
OK, I understand we should not need **any additional capability** - e.g. `CAP_SYS_NICE` , `CAP_SYS_ADMIN`, etc - when we're not root. 

But, I do not understand why we do require this capability - `CAP_NET_BIND_SERVICE` - unless we are using DHCP. Why is that capability explicitly required when the VMI is not executed as root @xpivarc ?

Reply:
I spoke with @xpivarc offline, and this is the gist of the explanation:
The launcher's pod golang based DHCP server is just a routine spawned from the launcher binary, and depending on whether or not the pod is running as root, it will feature the `CAP_NET_BIND_SERVICE` capability attached to the file itself.

That capability set will then be loaded when the file is executed. Hence, a non-root pod will have this capability attached to the file, which we need to enable in the pod. It is not very explicit, which means we should document it.

@AlonaKaplan

Reply:
I have to admit I don't understand what is written here and what is "the file".

But I have more basic question, if [this](https://github.com/kubevirt/kubevirt/blob/9ea85a2fe4a0fdd64d5d0b4f1e31e2fdfd732319/pkg/virt-controller/services/template.go#L1810) `if` is removed. Is there a scenario that will stop working?

Reply:
> I have to admit I don't understand what is written here and what is "the file".

The file is the `virt-launcher` binary.

> 
> But I have more basic question, if [this](https://github.com/kubevirt/kubevirt/blob/9ea85a2fe4a0fdd64d5d0b4f1e31e2fdfd732319/pkg/virt-controller/services/template.go#L1810) `if` is removed. Is there a scenario that will stop working?

@xpivarc would you chime in this one ?

Reply:
Sorry for the delay. 

Yes, the file is `virt-launcher` binary.

The `if` could be removed but we would then unnecessarily assign more capabilities to the launcher in the case of NonRoot. This is not desirable imo.

Reply:
@xpivarc which are the extra capabilities that must be added on non-root then ? The current status is misleading, and hard to reason about. 

IMO, we should use the minimal set of capabilities on each scenario - iiuc, we currently are not doing that.

Reply:
@xpivarc I will rephrase my question. In case of no  root vmi , can we add the CAP_NET_BIND only if `if requireDHCP(vmi) || haveSlirp(vmi) {
		capabilities = append(capabilities, CAP_NET_BIND_SERVICE)
	}`. Same as done for no root vmi?

Reply:
@maiqueb Currently we require `net_bind_service` `sys_ptrace`(for vtpm only temporary)

@AlonaKaplan yes and no. No, because you have the capability on the binary and therefore you are always trying to get it into the effective set. Therefore it would fail if you don't have the capability(Here it actually doesn't matter if it is root/less).
I say yes at the same time because **most** runtimes include the capability by default.

Reply:
I still don't fully understand.
Let's say we have a runtime that doesn't include the `net_bind_capability` by default.
Let's also say the vmi doesn't require DHCP or slirp.
Why in this case for root vmi we don't add the capability but for non-root vmi we do?
Would this scenario fail when using a root vmi?